cmd/res/llrp.device.profile.yaml

```diff
@@ -0,0 +1,289 @@
+name: "Device.LLRP.Profile"
+manufacturer: "Intel"
+model: "RFID Reader"
+labels:
+  - "LLRP"
+  - "RFID"
+description: "A generic device profile for readers that communicate over LLRP"
+
+deviceResources:
+  - name: "ReaderCapabilities"
+    description: "The capabilities of a reader"
+    properties:
+      value: { type: "String", readWrite: "R" }
+
+  - name: "ReaderConfig"
+    description: "Current configuration info of a reader"
+    properties:
+      value: { type: "String", readWrite: "RW" }
+
+  - name: "ROSpecID"
+    description: "Client-generated Reader Operation Specification Identifier"
```

Comment:
Correct. When a user submits an ROSpec, they choose an ID for it and embed it in the spec (the same thing is true of other LLRP spec types). Depending on its configuration, an LLRP device can embed this ID in messages it sends the user. For example, if your reader supports preemption, you might want to configure it to include the spec ID so you know which spec generated it. On the other hand, some readers don't even support having more than 1 spec on the device at a time, and so if you only ever use the same spec, you'd probably just disable getting the ID in the report. Even in that case, though, you might e.g. assign different spec IDs on a device-by-device basis, with the intent of just funneling all of their tag reports into a single system. It's a bit of a hack, I suppose, but the ID is fundamentally arbitrary, so it could be reasonable to embed some sort of meaning into it at an application level.

Regardless, if you want to enabled/disable/start/stop/delete a specific ROSpec, you have to refer to it by ID. Alternatively, you can often use "0" to operate on all specs at once. Somewhat annoyingly, I think the only way to query for spec IDs is to issue `GET_ROSPECS`, which doesn't accept parameters: it always returns all specs.